Improve test coverage of pkg/registry/util.go

jamoflaw opened the pull request:
**What this PR does / why we need it**:

Improves the test coverage of `pkg/registry/util.go`

**Special notes for your reviewer**:

Improves coverage of the package overall from 70.8% to 75.7%

**If applicable**:
- [ ] this PR contains user facing changes (the `docs needed` label should be applied if so)
- [x] this PR contains unit tests
- [x] this PR has been tested for backwards compatibility

jamoflaw committed:
Improve test coverage of pkg/registry/util.go

Signed-off-by: James Payne <jamoflaw@gmail.com>

jamoflaw commented on pkg/registry/util.go:
```diff
@@ -53,6 +53,7 @@ func ContainsTag(tags []string, tag string) bool {
 	return false
 }
 
+// GetTagMatchingVersionOrConstraint expects a sorted slice of semver tags (highest version first)
```
I thought it is worth adding a comment here as to what the method expects as its not clear that it expects this input to be pre-sorted

jamoflaw commented on pkg/registry/util_test.go:
```diff
@@ -273,3 +274,185 @@ func TestGenerateOCICreatedAnnotations(t *testing.T) {
+	assert.False(t, ContainsTag(tagList, "1.0.2"), "the tag 1.0.2 is not in the list")
+}
+
+func TestGetTagMatchingVersionOrConstraint(t *testing.T) {
```
This should test the scenarios outlined in the masterminds package https://pkg.go.dev/github.com/Masterminds/semver/v3@v3.3.0#section-readme as well as the explicit variants which are helm specific

mattfarina reviewed (changes requested):
Thank you for working on tests.

When we repeat the same sequence with different data there is a pattern for that. Could you please follow this pattern. There is an example [here](https://github.com/helm/helm/blob/31e22b9866af91e1a0ea2ad381798f6c5eec7f4f/pkg/cmd/pull_test.go#L52). If you search the code for `tests := []struct {` you'll find more examples.

jamoflaw committed:
Fixes review comments

Match standards within codebase

Signed-off-by: James Payne <jamoflaw@gmail.com>

jamoflaw commented:
> Thank you for working on tests.
> 
> When we repeat the same sequence with different data there is a pattern for that. Could you please follow this pattern. There is an example [here](https://github.com/helm/helm/blob/31e22b9866af91e1a0ea2ad381798f6c5eec7f4f/pkg/cmd/pull_test.go#L52). If you search the code for `tests := []struct {` you'll find more examples.

Updated as per review - I wasn't sure if there was a minimum for that style so I converted all of them - as it makes for a better reading experience anyway!

TerryHowe reviewed (approved):
/lgtm

Although TestGetTagMatchingVersionOrConstraint you could argue is testing NewConstraint in many cases.

gjenkins8 commented on pkg/registry/util_test.go:
```diff
@@ -273,3 +275,282 @@ func TestGenerateOCICreatedAnnotations(t *testing.T) {
+		isValid bool
+	}{
+		{
+			name:    "Valid url",
```
nit:
```suggestion
			name:    "Valid URI",
```

(A URL is a location you might send e.g. HTTP requests; a URI is just an identifier, which is true for OCI references)

gjenkins8 replied:
(and other places where we refer to OCI URIs as URLs)

jamoflaw replied:
Thanks @gjenkins8 - Hopefully thats covered now - I've updated all wording to be consistent I believe now!

jamoflaw committed:
Fixes review comments

Use correct acronym for OCI resources

Signed-off-by: James Payne <jamoflaw@gmail.com>

TerryHowe reviewed (approved):
/lgtm

TerryHowe commented:
```
< 							IsOCI						100.0%
< 							ContainsTag					100.0%
< 							GetTagMatchingVersionOrConstraint		94.1%
---
> 							IsOCI						0.0%
> 							ContainsTag					0.0%
> 							GetTagMatchingVersionOrConstraint		0.0%
```
